libs/execution_context/ebpf_link.c

```diff
@@ -20,16 +20,21 @@
  * 1. EBPF_LINK_STATE_INITIAL -> EBPF_LINK_STATE_ATTACHING - Program is being attached to a provider.
  * 2. EBPF_LINK_STATE_ATTACHING -> EBPF_LINK_STATE_ATTACHED - NmrRegisterClient returns success.
  * 3. EBPF_LINK_STATE_ATTACHING -> EBPF_LINK_STATE_DETACHING - Provider failed to attach.
- * 4. EBPF_LINK_STATE_ATTACHED -> EBPF_LINK_STATE_DETACHING - Program is being detached from a provider.
- * 5. EBPF_LINK_STATE_DETACHING -> EBPF_LINK_STATE_DETACHED - NmrDeregisterClient returns success.
+ * 4. EBPF_LINK_STATE_ATTACHED -> EBPF_LINK_STATE_LEGACY_MODE - Link is operating in legacy mode which prevents the link
+ * from detaching after the last user mode reference is released.
+ * 5. EBPF_LINK_STATE_LEGACY_MODE -> EBPF_LINK_STATE_DETACHING - Program is being detached from a provider.
+ * 6. EBPF_LINK_STATE_ATTACHED -> EBPF_LINK_STATE_DETACHING - Program is being detached from a provider.
+ * 7. EBPF_LINK_STATE_DETACHING -> EBPF_LINK_STATE_DETACHED - NmrDeregisterClient returns success.
  */
 typedef enum _ebpf_link_state
 {
-    EBPF_LINK_STATE_INITIAL,   ///< Program is not attached to any provider.
-    EBPF_LINK_STATE_ATTACHING, ///< Program is being attached to a provider.
-    EBPF_LINK_STATE_ATTACHED,  ///< Program is attached to a provider.
-    EBPF_LINK_STATE_DETACHING, ///< Program is being detached from a provider.
-    EBPF_LINK_STATE_DETACHED,  ///< Program is detached from a provider.
+    EBPF_LINK_STATE_INITIAL,     ///< Program is not attached to any provider.
+    EBPF_LINK_STATE_ATTACHING,   ///< Program is being attached to a provider.
+    EBPF_LINK_STATE_ATTACHED,    ///< Program is attached to a provider.
+    EBPF_LINK_STATE_LEGACY_MODE, ///< Link is operating in legacy mode. This prevents the link from detaching after the
```

Comment:
It's confusing that this is a separate state and doesn't have the word "INITIAL" or "ATTACHING" or "ATTACHED" or "DETACHING" or "DETACHED" in it.  Is it an orthogonal boolean or is this a subcategory of one of the other states?  ebpf_link.c line 641 seems to imply it's a subtype of ATTACHED, which if true then maybe EBPF_LINK_STATE_LEGACY_ATTACHED_MODE with the comment changing "operating" to "attached" or something like that.